kyleteeter/github-gists-app
# Display Gists

### Goal

The goal of this PR is to take the design from the last PR and add functionality. With these changes, the user will be able to search for a given GitHub username with Gists and see those results displayed. The user will also be able to click on different links to the Gist and see some basic info for each Gist. 

**_Mockup Design:_**

![Mockup](https://user-images.githubusercontent.com/29210839/147832542-780dd190-a1b9-4683-8fbe-79d6e659215a.png)

**_Design From this PR:_** 

![Screen Shot 2021-12-31 at 10 35 32 AM](https://user-images.githubusercontent.com/29210839/147832576-8c5809cb-efd0-414f-b821-fb70f3b276f3.png)

### Testing

To ensure proper functionality follow these steps:

_**Search Bar Test**_
1. Ensure default page load is as expected with no results.
2. Enter a GitHub username in the input and click "Get Gists". You should see the results display if the user has Gists.
3. Click the "Reset" button and ensure that all results are cleared and page looks as it did a first load.

**_Results Cards Test_**
1. While results are displayed ensure the design matches the mockup and is compatible with mobile. 
2. Ensure all links and buttons go to the expected destinations. 
3. Ensure the Gist language is displayed and in the correct color. 
4. Ensure that parameters are in place to handle if Gist description or Gist language is not available. 

### Notes

I am not sure if it would make more sense for me to break the SearchResults component down into various smaller components. It seems to be doing quite a few things in one component but they all seem related. Would appreciate some insight on this. Also, not sure if I set up the "No Search Results" correctly by using the isLoaded state. (Got the idea from [React docs](https://reactjs.org/docs/faq-ajax.html))
